realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -558,6 +568,13 @@ private void setValue(String fieldName, Object value) {
             RealmList<?> list = (RealmList<?>) value;
             setList(fieldName, list);
         } else {
+            // Handle Decimal128 and ObjectId in a special way since they might not be on the
+            // classpath
```

Comment:
I'm not sure about this comment, you added the Bson deps in the library gradle `implementation 'org.mongodb:bson:3.12.0'`